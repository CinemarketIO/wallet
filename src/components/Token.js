// @format
import React, { Component } from "react";
import styled from "styled-components";
import Modal from "react-modal";

import TransferModal from "./TransferModal";

const StyledToken = styled.div`
  margin: 1em;
  border: 1px solid #eee;
  border-radius: 1px;
  padding: 1em;
  background-color: #fafafa;
  text-align: center;
  width: 150px;
`;

const StyledImage = styled.img`
  display: inline-block;
  width: 100px;
  height: 100px;
`;

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)"
  }
};

const Token = props => (
  <StyledToken>
    <StyledImage width="100%" src={props.token && props.token.image} />
    <p>{(props.token && props.token.name) || "<No name given>"}</p>
    <p>
      {(props.token && props.token.description) || "<No description given>"}
    </p>
    {props.link ? (
      <a target="_blank" href={props.link}>
        Visit Website
      </a>
    ) : null}
    <button onClick={props.toggleModal(props.tokenId)}>Transfer</button>
    <Modal
      isOpen={props.modals[props.tokenId]}
      onRequestClose={props.toggleModal(props.tokenId)}
      style={customStyles}
      contentLabel="Example Modal"
      ariaHideApp={false}
    >
      <TransferModal
        tokenId={props.tokenId}
        from={props.account}
        contract={props.contract}
      />
    </Modal>
  </StyledToken>
);

export default Token;
