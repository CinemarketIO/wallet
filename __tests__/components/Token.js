// @format
import React from "react";
import renderer from "react-test-renderer";
import { ModalProvider } from "styled-react-modal";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

import Token from "../../src/components/Token";

describe("Token component renders the token correctly", () => {
  it("renders correctly", () => {
    const todo = { id: 1, done: false, name: "Buy Milk" };
    const token = {
      name: "Token",
      image: "https://example.com/image.jpg"
    };
    const tokenId = 123;
    const name = "nft name";
    const contract = "0xabc";
    const modals = {};
    const toggleModal = () => {};
    const account = "0xabc";
    const wrapper = shallow(
      <Token
        token={token}
        tokenId={tokenId}
        name={name}
        contract={contract}
        modals={modals}
        toggleModal={toggleModal}
        account={account}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
