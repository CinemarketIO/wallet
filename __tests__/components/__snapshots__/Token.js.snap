// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token component renders the token correctly renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Token
    account="0xabc"
    contract="0xabc"
    modals={Object {}}
    name="nft name"
    toggleModal={[Function]}
    token={
      Object {
        "image": "https://example.com/image.jpg",
        "name": "Token",
      }
    }
    tokenId={123}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          src="https://example.com/image.jpg"
          width="100%"
        />,
        <p>
          Token
        </p>,
        <p>
          &lt;No description given&gt;
        </p>,
        <a
          target="_blank"
        >
          Visit Website
        </a>,
        <button>
          Transfer
        </button>,
        <Modal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Example Modal"
          isOpen={false}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {
                "bottom": "auto",
                "left": "50%",
                "marginRight": "-50%",
                "right": "auto",
                "top": "50%",
                "transform": "translate(-50%, -50%)",
              },
            }
          }
        >
          <Connect(TransferModal)
            contract="0xabc"
            from="0xabc"
            tokenId={123}
          />
        </Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "src": "https://example.com/image.jpg",
          "width": "100%",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  display: inline-block;
  width: 100px;
  height: 100px;
",
            ],
          },
          "displayName": "styled.img",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "img",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Token",
        },
        "ref": null,
        "rendered": "Token",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<No description given>",
        },
        "ref": null,
        "rendered": "<No description given>",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Visit Website",
          "href": undefined,
          "target": "_blank",
        },
        "ref": null,
        "rendered": "Visit Website",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Transfer",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Transfer",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ariaHideApp": false,
          "bodyOpenClassName": "ReactModal__Body--open",
          "children": <Connect(TransferModal)
            contract="0xabc"
            from="0xabc"
            tokenId={123}
          />,
          "closeTimeoutMS": 0,
          "contentLabel": "Example Modal",
          "isOpen": false,
          "onRequestClose": undefined,
          "parentSelector": [Function],
          "portalClassName": "ReactModalPortal",
          "role": "dialog",
          "shouldCloseOnEsc": true,
          "shouldCloseOnOverlayClick": true,
          "shouldFocusAfterRender": true,
          "shouldReturnFocusAfterClose": true,
          "style": Object {
            "content": Object {
              "bottom": "auto",
              "left": "50%",
              "marginRight": "-50%",
              "right": "auto",
              "top": "50%",
              "transform": "translate(-50%, -50%)",
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "contract": "0xabc",
            "from": "0xabc",
            "tokenId": 123,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  margin: 1em;
  border: 1px solid #eee;
  border-radius: 1px;
  padding: 1em;
  background-color: #fafafa;
  text-align: center;
  width: 150px;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            src="https://example.com/image.jpg"
            width="100%"
          />,
          <p>
            Token
          </p>,
          <p>
            &lt;No description given&gt;
          </p>,
          <a
            target="_blank"
          >
            Visit Website
          </a>,
          <button>
            Transfer
          </button>,
          <Modal
            ariaHideApp={false}
            bodyOpenClassName="ReactModal__Body--open"
            closeTimeoutMS={0}
            contentLabel="Example Modal"
            isOpen={false}
            parentSelector={[Function]}
            portalClassName="ReactModalPortal"
            role="dialog"
            shouldCloseOnEsc={true}
            shouldCloseOnOverlayClick={true}
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
            style={
              Object {
                "content": Object {
                  "bottom": "auto",
                  "left": "50%",
                  "marginRight": "-50%",
                  "right": "auto",
                  "top": "50%",
                  "transform": "translate(-50%, -50%)",
                },
              }
            }
          >
            <Connect(TransferModal)
              contract="0xabc"
              from="0xabc"
              tokenId={123}
            />
          </Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "https://example.com/image.jpg",
            "width": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  display: inline-block;
  width: 100px;
  height: 100px;
",
              ],
            },
            "displayName": "styled.img",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "img",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Token",
          },
          "ref": null,
          "rendered": "Token",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<No description given>",
          },
          "ref": null,
          "rendered": "<No description given>",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Visit Website",
            "href": undefined,
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Visit Website",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Transfer",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Transfer",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ariaHideApp": false,
            "bodyOpenClassName": "ReactModal__Body--open",
            "children": <Connect(TransferModal)
              contract="0xabc"
              from="0xabc"
              tokenId={123}
            />,
            "closeTimeoutMS": 0,
            "contentLabel": "Example Modal",
            "isOpen": false,
            "onRequestClose": undefined,
            "parentSelector": [Function],
            "portalClassName": "ReactModalPortal",
            "role": "dialog",
            "shouldCloseOnEsc": true,
            "shouldCloseOnOverlayClick": true,
            "shouldFocusAfterRender": true,
            "shouldReturnFocusAfterClose": true,
            "style": Object {
              "content": Object {
                "bottom": "auto",
                "left": "50%",
                "marginRight": "-50%",
                "right": "auto",
                "top": "50%",
                "transform": "translate(-50%, -50%)",
              },
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "contract": "0xabc",
              "from": "0xabc",
              "tokenId": 123,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  margin: 1em;
  border: 1px solid #eee;
  border-radius: 1px;
  padding: 1em;
  background-color: #fafafa;
  text-align: center;
  width: 150px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
